apps/xprof_core/src/xprof_core_erlang_syntax.erl

```diff
@@ -121,7 +121,7 @@ fmt_exception(Class, Reason) ->
                                   SkipFun, PrettyFun, Encoding)]).
 
 fmt_term(Term) ->
-    fmt("~p", [Term]).
+    fmt("~w", [Term]).
```

Comment:
I think this is a bit simplistic solution (Gordian so to say). We still want to format printable strings as strings (eg `"foo"` not as `[102,111,111]`
I think the problem lies in what `jsone` thinks of encoding and what `io:format` does. (latin1 vs unicode)
I would dig a bit in the `unicode` module

Reply:
one option is to use `unicode:characters_to_binary/1` instead of `list_to_binary` in `fmt/2` so that at least it produces output that `jsone` can digest
```
2> jsone:encode([{item, list_to_binary([164])}]).
** exception error: bad argument
     in function  jsone_encode:escape_string/4
        called as jsone_encode:escape_string(<<"¤">>,
                                             [{object_members,[]}],
                                             <<"{\"item\":\"">>,
                                             {encode_opt_v2,false,
                                                            [{scientific,20}],
                                                            {iso8601,0},
                                                            string,0,0})
     in call from jsone:encode/2 (/Users/peter/git/xprof/_build/default/lib/jsone/src/jsone.erl, line 306)
3> jsone:encode([{item, unicode:characters_to_binary([164])}]).
<<"{\"item\":\"\\u00a4\"}">>
```

of course some integer lists will still be converted to strings although they are not intended to be like that, but that is an impossible problem to solve and we can only have heuristics.